enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,485 @@
+      toPort: 10259
+    cni:
+      cniIngressRules:
+      - description: ICMP
+        fromPort: -1
+        protocol: icmp
+        toPort: -1
+       # several rules omitted for brevity
+      - description: Service node ports (UDP)
+        fromPort: 30000
+        protocol: udp
+        toPort: 32767
+    vpc:
+      availabilityZoneSelection: Ordered
+      availabilityZoneUsageLimit: 3
+      cidrBlock: 10.0.0.0/16
+  region: us-east-2
+  s3Bucket:
+    name: openshift-bootstrap-data-capi-example-kbdpg
+    presignedURLDuration: 1h0m0s
+status:
+  networkStatus:
+    apiServerElb:
+      attributes: {}
+  ready: false
+```
+
+This manifest describes the network, load balancers, security group rules, etc. that will be used or provisioned for
+the target cluster. Control plane machines are provisioned by [AWSMachines][AWSMachines] & [CAPI Machines][CAPIMachines]
+manifests.
+
+##### Additional Infrastructure
+
+The Installer will generate any additional infrastructure that is needed but not handled by the CAPI provider, either
+because it is out of scope (e.g. IAM, DNS) or not adopted upstream (e.g. split-horizon load balancers). The Installer
+will provide hooks into the provisioning lifecycle that can be used to provision resources using direct SDK calls
+or other tooling.
+
+The AWS proof-of-concept implementation utilizes hooks defined in this interface:
+
+```go
+type CAPIInfraHelper interface {
+	// PreProvision is called before provisioning using CAPI controllers has begun.
+	// and should be used to create dependencies needed for CAPI provisioning,
+	// such as IAM roles or policies.
+	PreProvision(in PreProvisionInput) error
+
+	// ControlPlaneAvailable is called once cluster.Spec.ControlPlaneEndpoint.IsValid()
+	// returns true, typically after load balancers have been provisioned. It can be used
+	// to create DNS records.
+	ControlPlaneAvailable(in ControlPlaneAvailableInput) error
+}
+
+type PreProvisionInput struct{ clusterID string }
+type ControlPlaneAvailableInput struct{ *clusterv1.Cluster }
+```
+
+For AWS, IAM roles needed by the CAPA provider are created with `PreProvision` and DNS Records
+are created upon `ControlPlaneAvailable`. This interface would be implemented by each cloud provider
+and can be expanded as needed.
+
+##### Bootstrap Resources
+
+The Bootstrap Machine is created like the other control plane nodes, by defining a CAPI Machine resource.
+Accordingly, the bootstrap machine can be deleted by simply deleting the object. This works for all platforms:
+
+```go
+	if err := c.Delete(ctx, &clusterv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      capiutils.GenerateBoostrapMachineName(metadata.InfraID),
+			Namespace: capiutils.Namespace,
+		},
+	});
+```
+
+Further work needs to be done to determine how to delete additional bootstrap resources, such as SSH security group rules
+and the bootstrap ignition S3 bucket. These could be deleted either through updating the relevant manifests or created &
```

Comment:
IIUC at the moment the S3 ignition bucket is supposed to exist for the lifetime of a CAPI cluster, I know we want to fix this but we should probably call out that this is a difference right now and that it's something we aim to fix

Reply:
We might be able to disable the s3 integration after the first bootstrap is processed, and just use plain user-data for the rest of the workers nodes?

Reply:
Yeah that's possible. I had a discussion with the team prior to shutdown about putting some options in upstream to allow users to skip the S3 bucket option, I think there's valid use cases where CAPI should allow this so we could go down that route too.

Currently if you try to make the format plain, in AWS at least, you have to disable a cloud init encryption option which is not ideal

Reply:
IIRC it should be possible to disable the s3 bucket, and use plain user data, but we can double check

Reply:
Out of band discussion happening around this, @damdo is working on making this work upstream

Reply:
Hey yes. For reference here is the upstream PR I recently put up for that: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4700